public/app/features/transformers/partitionByValues/partitionByValues.ts

```diff
@@ -0,0 +1,112 @@
+import { map } from 'rxjs';
+
+import {
+  ArrayVector,
+  DataFrame,
+  DataTransformerID,
+  SynchronousDataTransformerInfo,
+  getFieldMatcher,
+} from '@grafana/data';
+import { getMatcherConfig } from '@grafana/data/src/transformations/transformers/filterByName';
+import { noopTransformer } from '@grafana/data/src/transformations/transformers/noop';
+
+import { partition } from './partition';
+
+export interface FrameNamingOptions {
+  /** whether to append to existing frame name, false -> replace */
+  append?: boolean; // false
+  /** whether to include discriminator field names, e.g. true -> Region=Europe Profession=Chef, false -> 'Europe Chef'  */
+  withFields?: boolean; // false
+  /** name/value separator, e.g. '=' in 'Region=Europe' */
+  separator1?: string;
+  /** name/value pair separator, e.g. ' ' in 'Region=Europe Profession=Chef' */
+  separator2?: string;
+}
```

Comment:
these are not currently exposed in the transformer UI, but allow you to choose how the resulting frames are named based on the found fields' unique values

Reply:
Interesting! Do you think customers want to give new names to the partitioned frames?